### PR TITLE
NF: document that `withCol` and `withQueue` executes in IO

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
I had some doubt about which function moves to IO dispatcher, whether it was launch method or withCol. I expect this documentation may save a minute to other devs too.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt [hunk 1]
```diff
@@ -58,8 +58,8 @@ object CollectionManager {
     var emulateOpenFailure = false
 
     /**
-     * Execute the provided block on a serial queue, to ensure concurrent access
-     * does not happen.
+     * Execute the provided block in the IO dispatchers on a serial queue.
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt:61]:
```suggestion
     * Execute the provided block on a serial background queue, to ensure concurrent access.
     * The background queue is run in a Dispatchers.IO context.
```

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt [hunk 2]
```diff
@@ -58,8 +58,8 @@ object CollectionManager {
...
+     * We use the serial queue to ensure concurrent access does not happen.
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt:62]:
```suggestion
```

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt [hunk 3]
```diff
@@ -88,6 +88,8 @@ object CollectionManager {
     /**
      * Execute the provided block with the collection, opening if necessary.
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt:89]:
```suggestion
     * Execute the provided block with the collection in a serialized background thread,
     opening if necessary.
```

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt [hunk 4]
```diff
@@ -88,6 +88,8 @@ object CollectionManager {
...
      *
+     * Execution is done in the IO dispatchers.
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt:91]:
```suggestion
     * Calls are serialized, and run in background Dispatchers.IO thread.
```

**Review by dae — COMMENTED** [review]
Just some minor wording bikeshedding.

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt (hunk 1 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt:61]: Involving `Dispatchers.IO` in brackets will make it clickable, i.e. `[Dispatchers.IO]`

**Comment by Arthur-Milchior** [comment]
I'm all for bikeshedding here, given that my goal was to improve documentation of a part of the code that seems to be non intuitive. (Or at least, was not to me at first)

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt (hunk 4 above)
Comment (dae) [reply AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt]: Sorry Arthur, I had a typo here: it should have read "run in a". Might as well add another [] at the same time.

File: AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt [hunk 5]
```diff
@@ -88,6 +90,8 @@ object CollectionManager {
     /**
      * Execute the provided block with the collection, opening if necessary.
      *
+     * Calls are serialized, and run in background Dispatchers.IO thread.
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/CollectionManager.kt:93]: nit
```suggestion
     * Calls are serialized, and run in background [Dispatchers.IO] thread.
```

**Review by BrayanDSO — APPROVED** [review]

**Pull request merged**